hubblestack/extmods/hubble_mods/grep.py

```diff
@@ -0,0 +1,177 @@
+# -*- encoding: utf-8 -*-
+"""
+Module for running grep command. Same can be used in both Audit/FDG
+
+Audit Example:
+---------------
+check_unique_id:
+  description: 'grep check'
+  tag: 'ADOBE-01'
+  implementations:
+    - filter:
+        grains: 'G@osfinger:CentOS*Linux-7'
+      hubble_version: '>3 AND <7 AND <8'
+      module: grep
+      items:
+        - args:
+            file: /etc/ssh/ssh_config
+            pattern: '"^MACs"'
+
+FDG Example:
+------------
+main:
+  description: 'grep check'
+  module: grep
+  args:
+    path: /etc/ssh/ssh_config
+    pattern: 'MACs'
+Mandatory parameters:
+    path - file path
+Multiple paths can be provided in a single implementation under attribute: "items"
+
+Note: Comparison logic is moved to comparators. Module will just invoke the grep command.
+"""
+
+import os
+import logging
+
+import hubblestack.extmods.module_runner.runner_utils as runner_utils
+from hubblestack.utils.hubble_error import HubbleCheckValidationError
+from salt.exceptions import CommandExecutionError
+
+log = logging.getLogger(__name__)
+
+
+def validate_params(block_id, block_dict, chain_args=None):
+    """
+    Validate all mandatory params required for this module
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param chain_args:
+        Chained argument dictionary, (If any)
+        Example: {'result': "/some/path/file.txt", 'status': True}
+
+    Raises:
+        AuditCheckValidationError: For any validation error
+    """
+    log.debug('Module: grep Start validating params for check-id: {0}'.format(block_id))
+
+    # fetch required param
+    mandatory_params = ['file', 'pattern']
+    error = {}
+    for param in mandatory_params:
+        param_val = runner_utils.get_param_for_module(block_id, block_dict, param, chain_args)
+        if not param_val:
+            error[param] = 'Mandatory parameter: %s not found for id: %s' % (param, block_id)
+
+    if error:
+        raise HubbleCheckValidationError(error)
+
+    log.debug('Validation success for check-id: {0}'.format(block_id))
+
+
+def execute(block_id, block_dict, chain_args=None):
+    """
+    For getting params to log, in non-verbose logging
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param chain_args:
+        Chained argument dictionary, (If any)
+        Example: {'result': "/some/path/file.txt", 'status': True}
+
+    returns:
+        tuple of result(value) and status(boolean)
+    """
+    log.debug('Executing grep module for id: {0}'.format(block_id))
+
+    # fetch required param
+    filepath = runner_utils.get_param_for_module(block_id, block_dict, 'file', chain_args)
+    pattern = runner_utils.get_param_for_module(block_id, block_dict, 'pattern', chain_args)
+    grep_args = runner_utils.get_param_for_module(block_id, block_dict, 'grep_args', chain_args)
```

Comment:
change grep_args, as we discussed yesterday

Reply:
done